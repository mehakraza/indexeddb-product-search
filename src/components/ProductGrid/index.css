.product-grid__btn-load-more {
  text-align: center;
  margin: 12px 46%;
  height: 32px;
}
