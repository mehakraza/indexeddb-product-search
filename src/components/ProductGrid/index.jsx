import React, { Component } from 'react'
import { Card, List, Image } from 'antd';
import { ProductModal } from '../ProductModal';
import './index.css';

export class ProductGrid extends Component {

  constructor(props) {
    super(props)

    this.state = {
      isModalVisible: false,
      selectedItem: {},
    }
  }

  setIsModalVisible = (item) => {
    this.setState({
      isModalVisible: !this.state.isModalVisible,
      selectedItem: item,
    });
  }

  handleCancel = () => {
    this.setIsModalVisible(false);
  };

  render() {
    const { Meta } = Card;
    const { isModalVisible, selectedItem } = this.state;
    const data = [];
    for(let i = 0; i < 10; i++) {
      data.push({
        title: "Weekday THURSDAY Jeans Slim Fit black",
        gtin: 2001007926858,
        gender: "female",
        sale_price: "39.95 EUR",
        price: "39.95 EUR",
        image_link: "https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.4.jpg",
        additional_image_link: [
          "https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@22.jpg",
          "https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@21.jpg",
          "https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@20.jpg",
          "https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@19.jpg",
          "https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@18.jpg",
        ],
      })
    };

    return (
      <>
      <List
        grid={{ gutter: 16, column: 4 }}
        pagination={{
          onChange: page => {
            console.log(page);
          },
          pageSize: 4,
        }}
        dataSource={data}
        renderItem={item => (
          <List.Item>
            <Card
              style={{ width: 300 }}
              cover={
                <Image
                  alt={item.title}
                  src={item.image_link}
                  preview={false}
                />
              }
              onClick={() => this.setIsModalVisible(item)}
            >
              <Meta
                title={item.title}
              />
              <div>GTIN: {item.gtin}</div>
              <div>Gender: {item.gender}</div>
              <div>Price: {item.price}</div>
              <div>Sale Price: {item.sale_price}</div>
            </Card>
          </List.Item>
          )}
        />
        <ProductModal
          item={selectedItem}
          isModalVisible={isModalVisible}
          handleCancel={this.handleCancel}
        />
      </>
    )
  }
}

export default ProductGrid
