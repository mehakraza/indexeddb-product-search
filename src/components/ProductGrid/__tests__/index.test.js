import React from 'react';
import { shallow } from 'enzyme';
import ProductGrid from '..';

const PROPS = {
  products: [
    {
      gtin: 2001007926858,
      title: 'Product Title 1',
      gender: 'female',
      price: '30.25',
      sale_price: '30.25',
      image_link: 'https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.41.jpg',
      additional_image_link: 'https://mosaic01.ztat.net/12.4.jpg, https://mosaic01.ztat.net/12.41.jpg, https://mosaic01.ztat.net/12.4.jpg',
    },
    {
      gtin: 2001007926859,
      title: 'Product Title 2',
      gender: 'female',
      price: '17',
      sale_price: '15',
      image_link: 'https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.42.jpg',
      additional_image_link: 'https://mosaic01.ztat.net/12.4.jpg, https://mosaic01.ztat.net/12.42.jpg, https://mosaic01.ztat.net/12.4.jpg',
    },
    {
      gtin: 2001007926860,
      title: 'Product Title 3',
      gender: 'male',
      price: '33.25',
      sale_price: '29.25',
      image_link: 'https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.43.jpg',
      additional_image_link: 'https://mosaic01.ztat.net/12.4.jpg, https://mosaic01.ztat.net/12.43.jpg, https://mosaic01.ztat.net/12.4.jpg',
    },
  ],
  populatingData: false,
  searching: false,
  showLoadMore: true,
  incrementPageNum: jest.fn(),
};

const getWrapper = (overrideProps = {}) => shallow(<ProductGrid {...PROPS} {...overrideProps} />);

describe('ProductGrid', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = getWrapper();
  });

  describe('rendering', () => {
    it('should render correctly when products are available', () => {
      expect(wrapper).toMatchSnapshot();
    });

     it('should render correctly when products are empty', () => {
      wrapper = getWrapper({ products: [] });
      expect(wrapper).toMatchSnapshot();
    });

    it('should render correctly when products are being loaded', () => {
      wrapper = getWrapper({ populatingData: true });
      expect(wrapper).toMatchSnapshot();
    });

    it('should render correctly when products are being searched', () => {
      wrapper = getWrapper({ searching: true });
      expect(wrapper).toMatchSnapshot();
    });

    it('should render correctly when showLoadMore is false', () => {
      wrapper = getWrapper({ showLoadMore: false });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
