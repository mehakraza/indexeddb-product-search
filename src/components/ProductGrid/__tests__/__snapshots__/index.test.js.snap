// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductGrid rendering should render correctly when products are available 1`] = `
<Fragment>
  <List
    dataSource={
      Array [
        Object {
          "additional_image_link": "https://mosaic01.ztat.net/12.4.jpg, https://mosaic01.ztat.net/12.41.jpg, https://mosaic01.ztat.net/12.4.jpg",
          "gender": "female",
          "gtin": 2001007926858,
          "image_link": "https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.41.jpg",
          "price": "30.25",
          "sale_price": "30.25",
          "title": "Product Title 1",
        },
        Object {
          "additional_image_link": "https://mosaic01.ztat.net/12.4.jpg, https://mosaic01.ztat.net/12.42.jpg, https://mosaic01.ztat.net/12.4.jpg",
          "gender": "female",
          "gtin": 2001007926859,
          "image_link": "https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.42.jpg",
          "price": "17",
          "sale_price": "15",
          "title": "Product Title 2",
        },
        Object {
          "additional_image_link": "https://mosaic01.ztat.net/12.4.jpg, https://mosaic01.ztat.net/12.43.jpg, https://mosaic01.ztat.net/12.4.jpg",
          "gender": "male",
          "gtin": 2001007926860,
          "image_link": "https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.43.jpg",
          "price": "33.25",
          "sale_price": "29.25",
          "title": "Product Title 3",
        },
      ]
    }
    grid={
      Object {
        "column": 4,
        "gutter": 16,
      }
    }
    loadMore={
      <Button
        className="product-grid__btn-load-more"
        onClick={[Function]}
      >
        Load more
      </Button>
    }
    loading={
      Object {
        "size": "large",
        "spinning": false,
        "tip": "Searching...",
      }
    }
    locale={
      Object {
        "emptyText": "No matching product found",
      }
    }
    renderItem={[Function]}
  />
  <ProductModal
    handleCancel={[Function]}
    isModalVisible={false}
    product={null}
  />
</Fragment>
`;

exports[`ProductGrid rendering should render correctly when products are being loaded 1`] = `
<Fragment>
  <List
    dataSource={
      Array [
        Object {
          "additional_image_link": "https://mosaic01.ztat.net/12.4.jpg, https://mosaic01.ztat.net/12.41.jpg, https://mosaic01.ztat.net/12.4.jpg",
          "gender": "female",
          "gtin": 2001007926858,
          "image_link": "https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.41.jpg",
          "price": "30.25",
          "sale_price": "30.25",
          "title": "Product Title 1",
        },
        Object {
          "additional_image_link": "https://mosaic01.ztat.net/12.4.jpg, https://mosaic01.ztat.net/12.42.jpg, https://mosaic01.ztat.net/12.4.jpg",
          "gender": "female",
          "gtin": 2001007926859,
          "image_link": "https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.42.jpg",
          "price": "17",
          "sale_price": "15",
          "title": "Product Title 2",
        },
        Object {
          "additional_image_link": "https://mosaic01.ztat.net/12.4.jpg, https://mosaic01.ztat.net/12.43.jpg, https://mosaic01.ztat.net/12.4.jpg",
          "gender": "male",
          "gtin": 2001007926860,
          "image_link": "https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.43.jpg",
          "price": "33.25",
          "sale_price": "29.25",
          "title": "Product Title 3",
        },
      ]
    }
    grid={
      Object {
        "column": 4,
        "gutter": 16,
      }
    }
    loadMore={
      <Button
        className="product-grid__btn-load-more"
        onClick={[Function]}
      >
        Load more
      </Button>
    }
    loading={
      Object {
        "size": "large",
        "spinning": true,
        "tip": "Loading data. Please wait...",
      }
    }
    locale={
      Object {
        "emptyText": "No matching product found",
      }
    }
    renderItem={[Function]}
  />
  <ProductModal
    handleCancel={[Function]}
    isModalVisible={false}
    product={null}
  />
</Fragment>
`;

exports[`ProductGrid rendering should render correctly when products are being searched 1`] = `
<Fragment>
  <List
    dataSource={
      Array [
        Object {
          "additional_image_link": "https://mosaic01.ztat.net/12.4.jpg, https://mosaic01.ztat.net/12.41.jpg, https://mosaic01.ztat.net/12.4.jpg",
          "gender": "female",
          "gtin": 2001007926858,
          "image_link": "https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.41.jpg",
          "price": "30.25",
          "sale_price": "30.25",
          "title": "Product Title 1",
        },
        Object {
          "additional_image_link": "https://mosaic01.ztat.net/12.4.jpg, https://mosaic01.ztat.net/12.42.jpg, https://mosaic01.ztat.net/12.4.jpg",
          "gender": "female",
          "gtin": 2001007926859,
          "image_link": "https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.42.jpg",
          "price": "17",
          "sale_price": "15",
          "title": "Product Title 2",
        },
        Object {
          "additional_image_link": "https://mosaic01.ztat.net/12.4.jpg, https://mosaic01.ztat.net/12.43.jpg, https://mosaic01.ztat.net/12.4.jpg",
          "gender": "male",
          "gtin": 2001007926860,
          "image_link": "https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.43.jpg",
          "price": "33.25",
          "sale_price": "29.25",
          "title": "Product Title 3",
        },
      ]
    }
    grid={
      Object {
        "column": 4,
        "gutter": 16,
      }
    }
    loadMore={
      <Button
        className="product-grid__btn-load-more"
        onClick={[Function]}
      >
        Load more
      </Button>
    }
    loading={
      Object {
        "size": "large",
        "spinning": true,
        "tip": "Searching...",
      }
    }
    locale={
      Object {
        "emptyText": "No matching product found",
      }
    }
    renderItem={[Function]}
  />
  <ProductModal
    handleCancel={[Function]}
    isModalVisible={false}
    product={null}
  />
</Fragment>
`;

exports[`ProductGrid rendering should render correctly when products are empty 1`] = `
<Fragment>
  <List
    dataSource={Array []}
    grid={
      Object {
        "column": 4,
        "gutter": 16,
      }
    }
    loadMore={
      <Button
        className="product-grid__btn-load-more"
        onClick={[Function]}
      >
        Load more
      </Button>
    }
    loading={
      Object {
        "size": "large",
        "spinning": false,
        "tip": "Searching...",
      }
    }
    locale={
      Object {
        "emptyText": "No matching product found",
      }
    }
    renderItem={[Function]}
  />
  <ProductModal
    handleCancel={[Function]}
    isModalVisible={false}
    product={null}
  />
</Fragment>
`;

exports[`ProductGrid rendering should render correctly when showLoadMore is false 1`] = `
<Fragment>
  <List
    dataSource={
      Array [
        Object {
          "additional_image_link": "https://mosaic01.ztat.net/12.4.jpg, https://mosaic01.ztat.net/12.41.jpg, https://mosaic01.ztat.net/12.4.jpg",
          "gender": "female",
          "gtin": 2001007926858,
          "image_link": "https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.41.jpg",
          "price": "30.25",
          "sale_price": "30.25",
          "title": "Product Title 1",
        },
        Object {
          "additional_image_link": "https://mosaic01.ztat.net/12.4.jpg, https://mosaic01.ztat.net/12.42.jpg, https://mosaic01.ztat.net/12.4.jpg",
          "gender": "female",
          "gtin": 2001007926859,
          "image_link": "https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.42.jpg",
          "price": "17",
          "sale_price": "15",
          "title": "Product Title 2",
        },
        Object {
          "additional_image_link": "https://mosaic01.ztat.net/12.4.jpg, https://mosaic01.ztat.net/12.43.jpg, https://mosaic01.ztat.net/12.4.jpg",
          "gender": "male",
          "gtin": 2001007926860,
          "image_link": "https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.43.jpg",
          "price": "33.25",
          "sale_price": "29.25",
          "title": "Product Title 3",
        },
      ]
    }
    grid={
      Object {
        "column": 4,
        "gutter": 16,
      }
    }
    loadMore={null}
    loading={
      Object {
        "size": "large",
        "spinning": false,
        "tip": "Searching...",
      }
    }
    locale={
      Object {
        "emptyText": "No matching product found",
      }
    }
    renderItem={[Function]}
  />
  <ProductModal
    handleCancel={[Function]}
    isModalVisible={false}
    product={null}
  />
</Fragment>
`;
